components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -197,12 +197,7 @@ public OAuth2ClientValidationResponseDTO validateClientInfo(String clientId, Str
                 regexp = registeredCallbackUrl.substring(OAuthConstants.CALLBACK_URL_REGEXP_PREFIX.length());
             }
 
-            if (regexp != null && callbackURI.matches(regexp)) {
-                validationResponseDTO.setValidClient(true);
-                validationResponseDTO.setApplicationName(appDO.getApplicationName());
-                validationResponseDTO.setCallbackURL(callbackURI);
-                return validationResponseDTO;
-            } else if (appDO.getCallbackUrl().equals(callbackURI)) {
+            if (regexp != null && callbackURI.matches(regexp) || appDO.getCallbackUrl().equals(callbackURI)) {
```

Comment:
Shall we define a private method like validateCallbackURI(String callbackURI, OAuthAppDO oauth) and move the callback validation logic there?

It would be much easy to refactor in the future when we support multiple callbacks in the future

Reply:
ok added a private method